packages/tasit-action/src/TasitAction.js

```diff
@@ -121,8 +170,9 @@ class TransactionSubscription extends Subscription {
           this.#txConfirmed = true;
         } else {
           if (this.#txConfirmed)
-            this.emitErrorEvent(
-              new Error(`Your message has been included in an uncle block.`)
+            this._emitErrorEvent(
+              new Error(`Your message has been included in an uncle block.`),
```

Comment:
Is this question stills open?

Reply:
Now it's more clear than before:

```javascript
const blockReorgOccurred =
    (receipt === null && this.#txConfirmations > 0) ||
    (receipt !== null && receipt.confirmations <= this.#txConfirmations);
```
Assuming that is called on each new block. On ordinary conditions, `receipt.confirmation` must be greater than last execution, otherwise the block was rejected. 
I think that condition cover well both self excludend scenarios.